tests/test_all_rules.py

```diff
@@ -1215,28 +1215,27 @@ def test_if_plugins_explicitly_defined(self):
 class TestAlertSuppression(BaseRuleTest):
     """Test rule alert suppression."""
 
-    @unittest.skipIf(PACKAGE_STACK_VERSION < Version.parse("8.8.0"),
-                     "Test only applicable to 8.6+ stacks for rule alert suppression feature.")
-    def test_group_length(self):
-        """Test to ensure the rule alert suppression group_by does not exceed 3 elements."""
-        for rule in self.production_rules:
-            if rule.contents.data.get('alert_suppression'):
-                group_length = len(rule.contents.data.alert_suppression.group_by)
-                if group_length > 3:
-                    self.fail(f'{self.rule_str(rule)} has rule alert suppression with more than 3 elements.')
-
     @unittest.skipIf(PACKAGE_STACK_VERSION < Version.parse("8.8.0"),
                      "Test only applicable to 8.8+ stacks for rule alert suppression feature.")
     def test_group_field_in_schemas(self):
         """Test to ensure the fields are defined is in ECS/Beats/Integrations schema."""
         for rule in self.production_rules:
-            if rule.contents.data.get('alert_suppression'):
-                group_by_fields = rule.contents.data.alert_suppression.group_by
+            rule_type = rule.contents.data.get('type')
+            if rule_type in ('query', 'threshold') and rule.contents.data.get('alert_suppression'):
```

Comment:
```suggestion
            if rule.contents.data.get('alert_suppression'):
```

Nit, is the additional check for `query` and `threshold` duplication here? I think this would be past the schema validation from  `validates_query_data` in rule.py that also checks for this.

Reply:
For now it's not. Once all `QueryRuleData` types support it, when we can remove this piece and add custom logic to check the fields.

Reply:
AH, yes good catch very good. Thanks!